src/components/App.jsx

```diff
@@ -1,75 +1,180 @@
-          redditUrl={process.env.REDDIT_URL}
-          appleAppStoreUrl={process.env.APPLE_APP_STORE_URL}
-          googlePlayUrl={process.env.GOOGLE_PLAY_URL}
-          handleAllTrackEvents={sendTrackEvent}
-        />
-      </div>
-    );
+  return (
+    <div>
+      <SiteHeader
+        logo={HeaderLogo}
+        loggedIn
+        username={username}
+        avatar={avatar}
+        logoAltText={configuration.SITE_NAME}
+        logoDestination={configuration.MARKETING_SITE_BASE_URL}
+        mainMenu={mainMenu}
+        userMenu={userMenu}
+        loggedOutItems={loggedOutItems}
+      />
+      <main>
+        <Switch>
+          <Route path="/u/:username" component={ConnectedProfilePage} />
+          <Route path="/error" component={ErrorPage} />
+          <Route path="/notfound" component={NotFoundPage} />
+          <Route path="*" component={NotFoundPage} />
+        </Switch>
+      </main>
+      <SiteFooter
+        siteName={configuration.SITE_NAME}
+        siteLogo={FooterLogo}
+        marketingSiteBaseUrl={configuration.MARKETING_SITE_BASE_URL}
+        supportUrl={configuration.SUPPORT_URL}
+        contactUrl={configuration.CONTACT_URL}
+        openSourceUrl={configuration.OPEN_SOURCE_URL}
+        termsOfServiceUrl={configuration.TERMS_OF_SERVICE_URL}
+        privacyPolicyUrl={configuration.PRIVACY_POLICY_URL}
+        facebookUrl={configuration.FACEBOOK_URL}
+        twitterUrl={configuration.TWITTER_URL}
+        youTubeUrl={configuration.YOU_TUBE_URL}
+        linkedInUrl={configuration.LINKED_IN_URL}
+        googlePlusUrl={configuration.GOOGLE_PLUS_URL}
+        redditUrl={configuration.REDDIT_URL}
+        appleAppStoreUrl={configuration.APPLE_APP_STORE_URL}
+        googlePlayUrl={configuration.GOOGLE_PLAY_URL}
+        handleAllTrackEvents={sendTrackEvent}
+      />
+    </div>
+  );
+}
+
+PageContent.propTypes = {
+  username: PropTypes.string.isRequired,
+  avatar: PropTypes.string,
+  ready: PropTypes.bool,
+  configuration: PropTypes.shape({
+    SITE_NAME: PropTypes.string.isRequired,
+    MARKETING_SITE_BASE_URL: PropTypes.string.isRequired,
+    SUPPORT_URL: PropTypes.string.isRequired,
+    CONTACT_URL: PropTypes.string.isRequired,
+    OPEN_SOURCE_URL: PropTypes.string.isRequired,
+    TERMS_OF_SERVICE_URL: PropTypes.string.isRequired,
+    PRIVACY_POLICY_URL: PropTypes.string.isRequired,
+    FACEBOOK_URL: PropTypes.string.isRequired,
+    TWITTER_URL: PropTypes.string.isRequired,
+    YOU_TUBE_URL: PropTypes.string.isRequired,
+    LINKED_IN_URL: PropTypes.string.isRequired,
+    GOOGLE_PLUS_URL: PropTypes.string.isRequired,
+    REDDIT_URL: PropTypes.string.isRequired,
+    APPLE_APP_STORE_URL: PropTypes.string.isRequired,
+    GOOGLE_PLAY_URL: PropTypes.string.isRequired,
+  }).isRequired,
+  intl: intlShape.isRequired,
+};
+
+PageContent.defaultProps = {
+  ready: false,
+  avatar: null,
+};
+
+const IntlPageContent = injectIntl(PageContent);
```

Comment:
The magic line that required the refactoring mentioned above - separating out PageContent lets us use `injectIntl` here.